backend/src/auth/auth42.strategy.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport'
+import { Strategy } from 'passport-oauth2';
+import { AuthService } from './auth.service';
+import { access } from 'fs/promises';
+
+@Injectable()
+export class Auth42Strategy extends PassportStrategy(Strategy, 'oauth') {
+  constructor(private authService: AuthService) {
+    super({
+      authorizationURL: 'https://api.intra.42.fr/oauth/authorize',
+      tokenURL: 'https://api.intra.42.fr/oauth/token',
+      clientID: process.env.FORTYTWO_APP_ID,
+      clientSecret: process.env.FORTYTWO_APP_SECRET,
```

Comment:
Should we short the environment variables name ?
Should we use `UUID` instead of `ID` ?
```suggestion
      clientID: process.env.FT_UUID,
      clientSecret: process.env.FT_SECRET,
```

Reply:
done